openssl-dynamic/src/main/c/jnilib.c

```diff
@@ -617,12 +645,12 @@ static jint JNI_OnLoad_netty_tcnative0(JavaVM* vm, void* reserved) {
 
     tcn_global_vm = vm;
     jint ret = netty_internal_tcnative_Library_JNI_OnLoad(env, packagePrefix);
-
-    if (packagePrefix != NULL) {
-      free(packagePrefix);
-      packagePrefix = NULL;
+    if (ret == JNI_ERR) {
+        free(packagePrefix);
```

Comment:
also set `staticPackagePrefix = NULL`?

Reply:
sure ... that said it should be `NULL` already. But I guess it can't harm and better safe than sorry =P